test/site-builder-test.js

```diff
@@ -333,12 +332,28 @@ describe('SiteBuilder', function() {
         .then(restoreLogs, restoreLogs);
     });
 
+    it('should build the site when the dest dir already exists', function() {
+      var handler = SiteBuilder.makeBuilderListener(webhook, builderConfig);
+
+      mySpawn.setDefault(mySpawn.simple(0));
+      captureLogs();
+      return filesHelper.createDir(path.join('dest_dir', 'foo'))
+        .then(function() {
+          return handler(incomingPayload).should.be.fulfilled;
+        })
+        .then(restoreLogs, restoreLogs);
+    });
+
     it('should create a builder that fails to build the site', function() {
       var handler = SiteBuilder.makeBuilderListener(webhook, builderConfig),
           errMsg = 'Error: failed to clone foo ' +
             'with exit code 1 from command: ' +
             'git clone git@github.com:18F/foo.git --branch 18f-pages';
 
+      // Ensure the output dir gets cleaned up. Though cloning the repo will
+      // fail, the builder will still create the output directory so that the
+      // build.log file will still be accessible to diagnose the error.
+      filesHelper.dirs.push(path.join('dest_dir', 'foo'));
```

Comment:
That was already happening by accident. Notice the test is already checking the contents of `dest_dir/foo/build.log`. The fixture was originally creating `dest_dir/foo` in `beforeEach`, but this PR takes that out, to make sure the new code is creating that directory instead.